controllers/status.go

```diff
@@ -0,0 +1,114 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	operatorv1alpha1 "github.com/kyma-project/api-gateway/apis/operator/v1alpha1"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type state int
+
+const (
+	Successful state = 0
```

Comment:
I think that ready might be better here, as `Successful` is not the name that will be on the CR in the end
```suggestion
	Ready state = 0
```